libs/ethers/src/lib/key-manager/+state/key-manager.service.ts

```
@@ -20,69 +23,81 @@ export class KeyManagerService {
-    this.store.setLoading(true);
     const wallet = EthersWallet.fromMnemonic(mnemonic);
-    this._setSigningKey(wallet);
-    wallet.encrypt(encryptionPassword).then(keyStore => {
-      const key = createKey({ensDomain, keyStore});
-      this.store.add(key);
-      this.store.setActive(key.id);
-      this.store.setLoading(false);
-    });
+    this._encryptAndStore(wallet, ensDomain, encryptionPassword);
   }
-  // create / encrypt / store / from private key
+  /** create / encrypt / store / from private key */
   importFromPrivateKey(ensDomain: string, privateKey: string, encryptionPassword: string) {
-    this.store.setLoading(true);
     const wallet = new EthersWallet(privateKey);
-    this._setSigningKey(wallet);
-    wallet.encrypt(encryptionPassword).then(keyStore => {
-      const key = createKey({ensDomain, keyStore});
-      this.store.add(key);
-      this.store.setActive(key.id);
-      this.store.setLoading(false);
-    });
+    this._encryptAndStore(wallet, ensDomain, encryptionPassword);
   }
 
-  // load key (retreive / decrypt, set into process memory)
-  unlockAndSetActive(key: Key, encryptionPassword: string) {
+  /** load key (retreive / decrypt, set into process memory) */
+  async unlockAndSetActive(key: Key) {
+    const ref = this.dialog.open(AskPasswordComponent, { width: '250px' })
+    const encryptionPassword = await ref.afterClosed().toPromise()
+    if (!encryptionPassword) throw new Error('No password provided');
+
     this.store.setLoading(true);
-    EthersWallet.fromEncryptedJson(key.keyStore, encryptionPassword).then(wallet => {
-      this._setSigningKey(wallet);
-      this.store.setActive(key.id);
+    try {
+      const wallet = await EthersWallet.fromEncryptedJson(key.keyStore, encryptionPassword)
+      this._activateKey(key.address, wallet);
       this.store.setLoading(false);
-    }).catch(error => {
+    }
+    catch(error) {
       console.warn(error);
```

You should rethrow the error is it's supposed to be display to the user.